In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import scipy.stats as stats
from random import *
%matplotlib inline

In [2]:
%run -i 'data_pipeline.py'

In [3]:
ratings_df

movie  critic_score  audience_score  \
0                 The Invisible Man            91              88   
1                      The Way Back            83              84   
2     Never Rarely Sometimes Always            99              72   
3                 Bad Boys for Life            77              96   
4                Sonic the Hedgehog            64              93   
...                             ...           ...             ...   
4582          Jayne Mansfield's Car            35              40   
4583              The Angels' Share            89              78   
4584              The Hunt (Jagten)            92              93   
4585                   Our Children            93              69   
4586                     Sightseers            85              62   

      release_date audience_tomatometer critic_tomatometer  score_difference  
0     Feb 28, 2020                Fresh              Fresh                -3  
1      Mar 6, 2020                Fresh              Fresh                 1  
2     Mar 13, 2020                Fresh              Fresh               -27  
3     Jan 17, 2020                Fresh              Fresh                19  
4     Feb 14, 2020                Fresh              Fresh                29  
...            ...                  ...                ...               ...  
4582  Sep 13, 2013               Rotten             Rotten                 5  
4583  Apr 12, 2013                Fresh              Fresh               -11  
4584  Jul 12, 2013                Fresh              Fresh                 1  
4585   Aug 2, 2013                Fresh              Fresh               -24  
4586  May 10, 2013                Fresh              Fresh               -23  

[4587 rows x 7 columns]

# Hypothesis Test ( Are means the Same)

##### Null Hypothesis: The mean of the critic scores equals the mean of the audience scores.

##### Alternative Hypothesis: The mean of the critic scores does not equals the mean of the audience scores.

##### Significance Level= 0.05

In [9]:
ratings_df['critic_score'].mean(),ratings_df['audience_score'].mean()

(62.06910834968389, 57.619359058207976)

In [34]:
stat,p_val=stats.ttest_ind(ratings_df['critic_score'],ratings_df['audience_score'])
stat,p_val

(8.519676403168509, 1.852933704152718e-17)

#### Reject Null. The mean of critic and audience scores is not equal.

# Hypothesis Test ( Are means within 5 points )

##### Null Hypothesis: The absolute value of the mean of the critic scores - the mean of the audience scores is greater than  =< 5

##### Alternative Hypothesis: The absolute value of the mean of the critic scores - the mean of the audience scores is greater than   >5

##### Significance Level= 0.05

In [13]:
ratings_df['score_difference'].mean()

-4.44974929147591

In [88]:
stat,p_val=stats.ttest_ind(ratings_df['critic_score']-5,ratings_df['audience_score'])
stat,p_val

(-1.0535330577432411, 0.2921245041007815)

#### Fail to Reject Null. The mean Audience score and Mean score are within 5 points of each other.

# Bayesian Test ( What is the probability the scores are within 5 points )

In [89]:
prob = round((abs(ratings_df['critic_score'] - ratings_df['audience_score'])<=5).mean() * 100, 2)
prob

16.68

####  The Audience score is within 5 points of the critic score 16.68% of the time.